agent/engine/docker_container_engine.go

```diff
@@ -59,6 +60,10 @@ const (
 	inspectContainerTimeout = 30 * time.Second
 	removeImageTimeout      = 3 * time.Minute
 
+	// Parameters for caching the docker auth for ECR
+	tokenCacheSize = 100
+	tokenCacheTTL  = 12 * time.Hour
```

Comment:
The token should tell you its own TTL.

Reply:
This part was mostly inherited from the original code, I think we can just get rid of this variable. As in the original code the cache will depend on the TTL to remove the expired token when calling `Get`[here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/async/lru_cache.go#L64). With the code change here, the expired token will be removed by `Delete` if `IsTokenValid` returned false when calling `Get`. Please correct me if I missed something.